sdk/aqueduct/dag_deltas.py

```diff
@@ -0,0 +1,329 @@
+import copy
```

Comment:
I had to move this into its own file so that I could import from utils. Conceptually, dag deltas should be the layer above dag.py.